server/server_test.go

```diff
@@ -2885,3 +2889,130 @@ func (m *mockResponseWriterConn) consumeQueryResultStream() ([]queryResultStream
 	}
 	return result, nil
 }
+
+func TestAuthenticationTLS(t *testing.T) {
```

Comment:
I can see value in testing the integration between the server, identifier, and authorizer. I'm just not sure we need to go all the way and actually bind the listener. I'm wary of the lack of cleanup, need to sleep(), and hardcoded server port. Is it possible to test the integration in the same we do for Bearer tokens? I'm willing to trust that the client certificates are being validated properly by the HTTP stack since the code that's enabling that is so small. WDYT?

Reply:
> Is it possible to test the integration in the same we do for Bearer tokens?

🤔 I'll think about this some more on Monday... there probably is a way... The question would be if what remains of this test would be worth it: Would it go beyond what [we test here](https://github.com/open-policy-agent/opa/pull/1149/files#diff-151881cb05585361d1f1ec36a8c0ad9aR20) in a way that still warrants the two test sets?